tests/rados/test_crash_daemon.py

```diff
@@ -0,0 +1,108 @@
+"""
+Module to verify health warning generated after crashing a ceph daemon
+"""
+import random
+import time
+
+from ceph.ceph_admin import CephAdmin
+from ceph.rados.core_workflows import RadosOrchestrator
+from tests.rados.monitor_configurations import MonConfigMethods
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """
+    # CEPH-83573855
+    Test to verify health warning generated after inducing a daemon crash
+    using crash mgr module
+    1. Choose an OSD daemon and a Mon daemon at random
+    2. Ensure cluster is in good state
+    3. Use crash common module to crash both the chosen daemons
+    4. Remove any existing or newly generated crash
+    """
+    log.info(run.__doc__)
+    config = kw["config"]
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    rados_obj = RadosOrchestrator(node=cephadm)
+    mon_obj = MonConfigMethods(rados_obj=rados_obj)
+
+    log.info("Test to verify crash warning upon inducing a crash for OSD and MON")
+
+    try:
+        # 1. Crash an OSD daemon and inject crash manually
+        out, _ = cephadm.shell(args=["ceph osd ls"])
+        osd_list = out.strip().split("\n")
+        osd_id = random.choice(osd_list)
+        # ensure chosen daemon is not already in crash state
+        health_detail, _ = cephadm.shell(["ceph health detail"])
+        assert f"osd.{osd_id} crashed" not in health_detail
+        assert rados_obj.crash_ceph_daemon(daemon="osd", id=osd_id, manual_inject=True)
+
+        # 2. Crash a MON daemon and let the crash show up automatically
+        mon_list = rados_obj.run_ceph_command("ceph status")["quorum_names"]
+        mon_id = mon_list[0]
```

Comment:
same here. These can be debug statements.

Reply:
done